code42/py42
# Bugfix/fix 2fa

### Description of Change ###

- Deprecates Py42MFARequiredError
- Adds `sdk.loginconfig.get_for_user(username)` method. 
- Catches `Py42Unauthorized` exception in `py42.sdk.from_local_account` and provides more helpful exception messages.


### Testing Procedure ###

SSO User test:
- Attempt to initialize the SDKClient using `py42.sdk.from_local_account()` with a user configured for SSO auth.
- A `Py42Error` should be raised with text: ```SSO users are not supported in `from_local_account()`.```

Local & 2FA user test:
- Attempt to initialize the SDKClient using `py42.sdk.from_local_account()` with a local user (both with 2FA enabled and without), but use incorrect credentials and/or TOTP token.
- A `Py42UnauthorizedError` should be raised with text: ```SDK initialization failed, double-check username/password, and provide two-factor TOTP token if Multi-Factor Auth configured for your user. User LoginConfig: {login_type}``` where `login_type` matches that of the user you used (either `LOCAL` or `LOCAL_2FA`). 

Get Login Configuration method test:
- Initialize the SDKClient from the `SDKClient.from_local_account()` classmethod (this doesn't check credentials on initialization. Use incorrect credentials as `sdk.loginconfig.get_for_user()` should work unauthenticated:
```
from py42.sdk import SDKClient
sdk = SDKclient.from_local_account("{console_url}", "{username}", "{bad_password}")
```

- Call `sdk.loginconfig.get_for_user("{username"})`
- the `loginType` key from the response data should match that of the user login type requested (`LOCAL`, `LOCAL_2FA`, or `CLOUD_SSO`).

### PR Checklist ###
Did you remember to do the below?
